tox.ini

```diff
@@ -10,7 +10,16 @@ setenv =
     LC_CTYPE = en_US.UTF-8
 deps = -r{toxinidir}/dev-requirements.txt
 commands = py.test --timeout 300 []
-install_command = python -m pip install {opts} {packages}
+# Make sure the version of pip we're about to test is
+# not itself being used for setting up the environment.
```

Comment:
I find it completely abnormal to be using the code to be tested for setting up the environment itself, it should not be trusted.

Furthermore, if it fails during the tox env setup phase, instead of failing during the tests, the information you'll get from the `install_command` log are going to be else useful for identifying what went wrong then the results of a testsuite run.

And if the `install_command` does not return error that does not necessarily mean the environment was correctly setup: if a subtle bug was introduced in the code (like if suddenly pre-releases are being used when not asked for). How can you trust the testsuite then?

IMHO the whole process should be as reproducible as possible, and that means using a stable version of pip to set up the environment so the testsuite always as a chance to be run.